src/features/common/Layout/Navbar/microComponents/NavbarItems.tsx

```diff
@@ -0,0 +1,40 @@
+import type { HeaderItem, NavbarItemHeaderKey } from '../defaultTenantConfig';
+
+import { useTenant } from '../../TenantContext';
+import UserProfileButton from './UserProfileButton';
+import { useState } from 'react';
+import NavbarItemGroup from './NavbarItemGroup';
+
+const NavbarItems = () => {
+  const { tenantConfig } = useTenant();
+  const headerItems = tenantConfig.config.header.items;
```

Comment:
I think it may make sense to apply the translation logic here only, and send down translated text to the child components as props.

The logic of the child components will be greatly simplified.